stan/math/prim/fun/unit_vector_free.hpp

```diff
@@ -20,9 +21,11 @@ namespace math {
  * @return Unit vector of dimension K considered "free"
  */
 template <typename EigVec, require_eigen_vector_t<EigVec>* = nullptr>
-inline decltype(auto) unit_vector_free(EigVec&& x) {
-  check_unit_vector("stan::math::unit_vector_free", "Unit vector variable", x);
-  return std::forward<EigVec>(x);
+inline auto unit_vector_free(EigVec&& x) {
+  const auto& x_ref = to_ref(x);
```

Comment:
```suggestion
  auto&& x_ref = to_ref(std::forward<EigVec>(x));
```

Reply:
On a side note, should we be using `auto&&` instead of `const auto&` when we do to_ref()? tmk `auto&&` evaluates to any reference type. Though I'm indifferent about it

https://godbolt.org/z/aYG5na

Reply:
Actually it does not matter, unless you need the reference to be non-const. I am just more used to const&.